upgrade to kubernetes v1.7

andrewsykim opened the pull request:
Working off of https://github.com/kubernetes-digitalocean-terraform/kubernetes-digitalocean-terraform/pull/39. This is a seperate PR to only upgrade to kubernetes v1.7 

On a side note I'd like to increase the default size of the masters, I find that only having 1 core usually overloads the servers.

Commit:
upgrade to kubernetes v1.7

5tefan commented:
Was getting stuck on null_resource.deploy_dns_addon and null_resource.deploy_microbot due tp master refusing connections and couldn't even ssh in. Agree with you, seems to work with `size_master` set to "1gb". 

Haven't finished looking through the k8s changelog yet, but [1]: 
> NetworkPolicy has been promoted from extensions/v1beta1 to the new networking.k8s.io/v1 API group
Does that mean we should remove [2]: `--runtime-config=extensions/v1beta1/networkpolicies=true`?


[1] https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#v170
[2] https://github.com/kubernetes-digitalocean-terraform/kubernetes-digitalocean-terraform/blob/6e44c3ba66a32c19fdea9b20feb7fcefbdbd0d1d/01-master.yaml#L87

5tefan committed:
512mb master becoming unresponsive

Master becomes unresponsive and cluster fails to deploy with 512mb
nodes. Change default master_size to 1gb size.

5tefan commented:
Noticing from the 1.7.0 changelog [1] that our dns addon is a bit out of date.

We're on kubedns 1.8 and dnsmasq 1.4, upstream k8s addon [2] is at 1.14.2 and has dns-sidecar (not sure what that is). 

Maybe updating dns stuff is another PR.

[1] https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#networking-1
[2] https://github.com/kubernetes/kubernetes/pull/45684/files#diff-029d9fc1138abb853c9e8f9233b77de5

5tefan commented:
#38 requests v1.6+ so this fixes #38.

5tefan committed:
removes network policies runtime config

NetworkPolicy has been promoted from extensions/v1beta1
to the new networking.k8s.io/v1 API group. [1]

[1] https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#network
